backend/cmd/headlamp.go

```diff
@@ -132,6 +136,77 @@ func StartHeadlampServer(config *HeadlampConfig) {
 
 	// Configuration
 	r.HandleFunc("/config", clientConf.getConfig).Methods("GET")
+	r.HandleFunc("/authtype", getClusterAuthType).Queries("cluster", "{cluster}").Methods("GET")
+
+	state := "app"
+	var oauthConfig oauth2.Config
+	var verifier *oidc.IDTokenVerifier
+	ctx := context.Background()
+
+	r.HandleFunc("/oidc", func(w http.ResponseWriter, r *http.Request) {
+		cluster := r.URL.Query().Get("cluster")
+		authConfig, err := GetClusterAuthConfig(cluster)
+		if err != nil {
+			log.Fatal(err)
+		}
+		provider, err := oidc.NewProvider(ctx, authConfig.IdpIssuerURL)
+		if err != nil {
+			log.Fatal(err)
+		}
+		oidcConfig := &oidc.Config{
+			ClientID: authConfig.ClientID,
+		}
+		verifier = provider.Verifier(oidcConfig)
+		oauthConfig = oauth2.Config{
+			ClientID:     authConfig.ClientID,
+			ClientSecret: authConfig.ClientSecret,
+			Endpoint:     provider.Endpoint(),
+			RedirectURL:  fmt.Sprintf("http://127.0.0.1:%1s/callback", config.port),
+			Scopes:       []string{oidc.ScopeOpenID, "profile", "email", "groups"},
+		}
+		http.Redirect(w, r, oauthConfig.AuthCodeURL(state), http.StatusFound)
+	}).Queries("cluster", "{cluster}")
+
+	r.HandleFunc("/callback", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
hmm, yeah but with the change we will have to register this `oidc-cb` or `oidc-callback` as the auth callback endpoint instead of /callback.
Also yeah the callback can be called with POST method that will be used to post the refresh token and fetch a new token with that